File: Sort Algorithms/Selection Sort Algorithm/SelectionSortAlgorithm.cpp
#include <iostream>
#include <vector>

using namespace std;

void selectionSort(vector<int>& nums) {
	for (int i = 0; i < nums.size(); ++i) {
		int min_index = i;
		for (int j = i; j < nums.size(); ++j) {
			if (nums[j] < nums[min_index])
				min_index = j;
		}
		int temp = nums[i];
		nums[i] = nums[min_index];
		nums[min_index] = temp;
 	}
}

void print(vector<int>& nums) {
	cout << "[";
	for (int i = 0; i < nums.size(); ++i)
		cout << nums[i] << ((i != nums.size() - 1)? ", ":"");
	cout << "]" << endl;
}

int main(int argc, char* argv[]) {
	vector<int> nums = {32, -5, 0, 43, 72, 8, 9, 1, 32};
	print(nums);
	selectionSort(nums);
	print(nums);
	return 0;
}

File: Sort Algorithms/Selection Sort Algorithm/SelectionSortAlgorithm.py
def selectionSort(nums):
    for i in range(len(nums)):
        min_index = i
        for j in range(i, len(nums)):
            if nums[j] < nums[min_index]:
                min_index = j
        nums[i], nums[min_index] = nums[min_index], nums[i]

nums = [41, 32, 16, 0, 1, 47, 100, 329, 50]
selectionSort(nums)
print(nums)
